druid/src/contexts.rs

```diff
@@ -528,11 +539,29 @@ impl<'a, 'b: 'a> PaintCtx<'a, 'b> {
         let mut child_ctx = PaintCtx {
             render_ctx: self.render_ctx,
             base_state: self.base_state,
+            z_ops: Vec::new(),
             window_id: self.window_id,
             focus_widget: self.focus_widget,
             region: region.into(),
         };
-        f(&mut child_ctx)
+        f(&mut child_ctx);
+        self.z_ops.append(&mut child_ctx.z_ops);
+    }
+
+    /// Allows to specify order for paint operations.
+    ///
+    /// Larger `z_idx` indicate that an operation will be executed later.
```

Comment:
It feels like this is documenting the implementation more than the interface, but I've had trouble coming up with anything better, so I'm happy enough with this; at least it is clear.

edit: also the comment says `z_idx` but the code says `z_index` 🤷‍♀